src/contexts/TransactionContext.tsx

```diff
@@ -65,6 +69,10 @@ export function TransactionContextProvider(props: TransactionContextProviderProp
     payloadHex: null
   });
 
+  useEffect((): void => {
+    account && dispatchTransaction(TransactionActionCreators.setSenderAccount(account.address));
```

Comment:
```suggestion
    dispatchTransaction(TransactionActionCreators.setSenderAccount(account ? account.address : null));
```

If `account` is set back to `null` (or in general something undefined) we probably want the reducer to be updated as well.

Reply:
at the moment this scenario is not possible since you choose an account you don't have the possibility to clean the account, but as i suppose this will be possible in the refactored Sender component this might be valid.

Reply:
The fact that something is not **currently** clickable (or we believe it is not) does not justify taking shortcuts in code correctness.
1. Setting it to `null` is strictly better than not doing anything.
2. The code change is trivial
3. Assumptions like that quickly get outdated.
Hence I'd be always more in favour "future proof" behaviors.

Thanks for addressing that!

Reply:
of course, that's why i implemented your proposal without any questioning.